satpy/tests/test_demo.py

```diff
@@ -192,3 +192,47 @@ def test_ahi_partial_download(self):
         from tempfile import gettempdir
         files = download_typhoon_surigae_ahi(base_dir=gettempdir(), segments=[4, 9], channels=[1, 2, 3])
         assert len(files) == 6
+
+
+class _FakeZipRequest:
+    """Fake object to act like a requests return value when downloading a zip file."""
+
+    def __init__(self, zip_dir_name):
+        self._dir_name = zip_dir_name
+        self.headers = {}
+
+    def _get_fake_viirs_sdr_zip(self):
+        import zipfile
+        filelike_obj = io.BytesIO()
+        with zipfile.ZipFile(filelike_obj, mode="w") as z_file:
+            for i in range(6 * 3):
+                v_fn = f"{i}.h5"
```

Comment:
Could these files be named with the real file names? This way you could then do more than check the number of files in the `test_download` method by checking that the files are what we expect.

Reply:
I could, but if I'm checking that the files are as we expect in the test then why not keep them like this and still add the check?